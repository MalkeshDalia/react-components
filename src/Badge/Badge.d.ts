import { PureComponent } from 'react';

export type BadgeProps = {
  label: any,
  active?: boolean,
  className?: string,
  style?: object
}

export default class Badge extends PureComponent<BadgeProps> {}
