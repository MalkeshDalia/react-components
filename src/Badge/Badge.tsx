import * as React from 'react';
import cx from '../utils/classnames';
import { props, t, ReactNode } from '../utils';
import FlexView from 'react-flexview';

export type BadgeProps = {
  /** the descriptive content of the badge */
  label: React.ReactNode,
  /** tells if the badge is active (for styling purposes) */
  active?: boolean,
  /** an optional class name to pass to top level element of the component */
  className?: string,
  /** an optional style object to pass to top level element of the component */
  style?: React.CSSProperties
};

export const Props = {
  label: ReactNode,
  active: t.maybe(t.Boolean),
  className: t.maybe(t.String),
  style: t.maybe(t.Object)
};

@props(Props)
export default class Badge extends React.PureComponent<BadgeProps> {

  render() {
    const { label, active, className: _className, style } = this.props;
    const className = cx('badge', { active }, _className);

    return (
      <FlexView vAlignContent='center' hAlignContent='center' {...{ className, style }}>
        <span className='badge-label'>{label}</span>
      </FlexView>
    );
  }
}
