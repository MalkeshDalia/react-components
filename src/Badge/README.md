# Badge

A small descriptive element to attach to other UI components

## Props
|Name|Type|Default|Description|
|----|----|-------|-----------|
| **label** | <code>ReactChildren</code> |  | **required**. The descriptive content of the badge |
| **active** | <code>Boolean</code> |  | *optional*. Tells if the badge is active (for styling purposes) |
| **className** | <code>String</code> |  | *optional*. Additional `className` for wrapper element |
| **style** | <code>Object</code> |  | *optional*. Inline-style overrides for wrapper element |