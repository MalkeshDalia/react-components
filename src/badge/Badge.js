import React from 'react';
import cx from 'classnames';
import { skinnable, props, t } from '../utils';
import FlexView from 'react-flexview';

export const Props = {
  label: t.ReactChildren,
  active: t.maybe(t.Boolean),
  className: t.maybe(t.String),
  style: t.maybe(t.Object)
};

/** A small descriptive element to attach to other UI components
 * @param label - the descriptive content of the badge
 * @param active - tells if the badge is active (for styling purposes)
 */
@skinnable()
@props(Props)
export default class Badge extends React.PureComponent {

  getLocals() {
    const { label, active, className, style } = this.props;

    return {
      label,
      style,
      className: cx('badge', { active }, className)
    };
  }

  template({ label, className, style }) {
    return (
      <FlexView vAlignContent='center' hAlignContent='center' {...{ className, style }}>
        <span className='badge-label'>{label}</span>
      </FlexView>
    );
  }
}
