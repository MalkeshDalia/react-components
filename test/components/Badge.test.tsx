import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Badge from '../../src/Badge';

describe('Badge', () => {

  it('renders correctly', () => {
    const component = renderer.create(
      <Badge label='42' />
    );
    expect(component).toMatchSnapshot();
  });

  it('computes className', () => {
    const badge = shallow(
      <Badge label={42} className='awesome-class' />
    );
    expect(badge.hasClass('badge'));
    expect(badge.hasClass('awesome-class'));
    expect(!badge.hasClass('active'));
  });

  it('computes className', () => {
    const badge = shallow(
      <Badge label={42} className='awesome-class' />
    );
    expect(badge.hasClass('badge'));
    expect(badge.hasClass('awesome-class'));
    expect(badge.hasClass('active'));
  });

});
