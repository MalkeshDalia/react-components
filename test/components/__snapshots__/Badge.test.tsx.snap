// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<div
  className="react-flex-view align-content-center justify-content-center badge"
  style={
    Object {
      "MozBoxFlex": "0 1 auto",
      "WebkitBoxFlex": "0 1 auto",
      "WebkitFlex": "0 1 auto",
      "flex": "0 1 auto",
      "msFlex": "0 1 auto",
    }
  }
>
  <span
    className="badge-label"
  >
    42
  </span>
</div>
`;
